pkg/tfbridge/validate_input_types.go

```diff
@@ -274,3 +285,26 @@ func (v *PulumiInputValidator) ValidateInputs(resourceToken tokens.Type, inputs
 
 	return nil
 }
+
+// ValidateConfig will validate the provider config against the pulumi schema. It will
+// return a list of type failures if any are found
+func (v *PulumiInputValidator) ValidateConfig(inputs resource.PropertyMap) []TypeFailure {
+	// The inputs that are provided to `CheckConfig` also include pulumi options
+	// so make sure we are only checking against properties in the config
+	inputsToValidate := resource.PropertyMap{}
+	for _, k := range inputs.StableKeys() {
+		if _, ok := v.schema.Config.Variables[string(k)]; ok {
+			inputsToValidate[k] = inputs[k]
+		}
+	}
+	failures := v.validatePropertyMap(
+		inputsToValidate,
+		v.schema.Config.Variables,
+		resource.PropertyPath{},
+	)
+	if len(failures) > 0 {
+		return failures
+	}
+
+	return nil
```

Comment:
Distinguishing between `[]TypeFailure{}` and `[]TypeFailure(nil)` is an anti-pattern.

```suggestion
	return v.validatePropertyMap(
		inputsToValidate,
		v.schema.Config.Variables,
		resource.PropertyPath{},
	)
```

In general, we should strive to behave the same when given an empty slice or a `nil` slice.